docs/content/misc/faq.ngdoc

```diff
@@ -65,6 +58,44 @@ is also deprecated but we only remove the feature once it's removed from jQuery
 jqLite and jQuery. One such example is the `bind` method, deprecated in favor of `on` but unlikely to be removed
 from jqLite any time soon.
 
+#### What is the version compatibility between AngularJS main and optional modules?
+
+AngularJS code is separated into a main module ("angular"), and a few different optional modules
+("angular-animate", "angular-route" etc) that are dependant on the main module.
+When a new AngularJS version is released, all modules are updated to the new version.
+This means that the main module and the optional modules must always have the exact same version,
+down the patch number, otherwise your application might break.
+
+Therefore you must always explicitly lock down your dependencies, for example in the package.json,
+the following ensures that "angular" and "angular-animate" are always updated to the same version.
+
+```
+{
+  "angular": "~1.6.0",
+  "angular-animate": "~1.6.0"
+}
+```
+
+#### How does AngularJS ensure code quality and guard against regressions?
+
+When adding new code to branches of AngularJS, have a very stringent commit policy:
+
+- Every commit must contain tests and documentation updates alongside the code changes and that all the
+tests must pass;
+- Commit messages must be written in a specific manner that allows us to parse them and extract the changes
+for release notes.
+
+The AngularJS code base has a very large set of unit tests and end-to-end tests. This means that a breaking change will require one or more tests to be changed to allow the
+tests to pass. So when a commit includes tests that are being removed or modified, this is a flag that the
+code might include a breaking change. When reviewing the commit we can then decide whether there really is
+a breaking change and if it is appropriate for the branch to which it is being merged. If so, then we
+require that the commit message contains an appropriate breaking change message.
+
+Additionally, commits are periodically synced to Google where we test it against applications using
+the test suites of these applications. This allows us to catch regressions
+quickly before a release. We've had a pretty good experience with this setup. Only bugs that affect features
+not used at Google or without sufficient test coverage, have a chance of making it through.
```

Comment:
I assume they are meant to inspire confidence in the stability of AngularJS (I've just reworded them from their previous state)